### DISCUSSION TITLE
Support downloading specific files

### DISCUSSION THREAD

**Discussion by Zagan66** [community/community#13304] [Other Feature Feedback, Questions, & Ideas]
I get sent here a lot trying to get files but it's usually a dead end because I can't download anything.

WHY THE HELL DON'T YOU HAVE A SIMPLE "Download" button so that it's a million times easier to download files?
I would never upload anything here, it's just way too confusing. And most of this stuff has no instructions either. Like I'm supposed to magically know how to install and use all these loose files I've never even seen before.

Please make this site more user friendly, like for people who don't know all this programming stuff.

**Comment by attomos** [comment]
Maybe, you can start learning about Git and GitHub from [Quickstart](https://docs.github.com/en/get-started/quickstart/hello-world) documentation.

And downloading files from GitHub is pretty easy even if you don't have Git setup.

I hope this screen recording can help you to download what you want.
https://user-images.githubusercontent.com/684347/159675793-6dacd1b7-9fa7-4e7d-ae99-5c5bc63e96ca.mov

**Reply by ahmedjsofficial** [reply]
Oh wow that great! it helps a lot

**Comment by zettca** [comment]
GitHub's main purpose is for _storing_ and sharing **code**, not installers or executables. That's why the primary button in a project page is to download the code, and the main view is the repository's folder structure.
Yes, GitHub has a nice releases feature (under a project's `/releases`) where you can store extra files that are not part of the codebase (i.e. installers), but that's a secondary feature.

If you want better documentation or installing instructions, ask the project owners to improve the documentation or create a "proper" download website, not GitHub.

An analogy, in case I wasn't clear enough:
When you download a buggy unfinished game from Steam, you should complain to the game devs, not Steam itself. Steam's job is to publish and making it easier to install/start the game.

**Reply by ahmedjsofficial** [reply]
what the hell is this

**Reply by zettca** [reply]
what?

**Comment by RAM237** [comment]
Your anger is probably targeted to some projects poor documentation, not github itself :)

Chill man, github is generally very friendly, couple of years ago I studied Python with people who had 0 level of programming skills, and I guess 100% of them managed to work with github and submit their projects with no issues.

Take care!

**Comment by ghost** [comment]
I saw the post title and just HAD to shove my nose in.

tl;dr: I agree with OP, with some expansions. I will seem to have a habit of reiterating _ad nauseam_ to some people. Sorry. (Plus: Sarcasm warning!)

Unfortunately, as with most things social, it seems you're either 'in the clique', or an outcast.  But this depends on the prevailing culture at the time and different project certainly have different cultures surrounding them. Some to the detriment of the project.

IMHO, a more informative, intuitive and inclusive interface would be a step forwards, but then how would the 1337 look cool if the plebs are allowed into the club? OP makes it sound like just anyone should be able to access or browse GitHub

I have to agree with the OP in that there is a lot of presumed knowledge and understanding of GitHub (and it's linked functionality) and any niche-specific language is going to be exclusionary.

Like teenagers creating new words and phrases to exclude the 'uncool' parents, or a company 'training' you on proprietary systems - It's ultimately fruitless and doesn't do anything for productivity or communication outside of that niche environment.  As a previously RM-certified engineer, it was a useless set of qualifications that didn't help me progress in my career.

You can't 'un-know' something, this makes it hard for those who 'do know' to empathise or sympathise with those who 'don't know'.

Whilst RTFM is a valid method for gaining information, sometimes the manual is 4 lines of nothing text.

Anger at things that aren't quite right or fair is perfectly normal.  It is truly frustrating, especially as a programmer who prefers to not use GitHub.  I find the Git-specific language both exclusionary and inhibitory... I have to learn ANOTHER third-party toy in order to access projects on GitHub. Sorry, but there are better and more important things to learn.

@RAM237 : Programming skills <> GitHub skills, which is painfully obvious to anyone who frequents GitHub. ( Also, 2 of your friends, without coding skills, could use GitHub?  I think they're still on GitHub and there's definitely more than 2 of them lol jk. It's hard to admit you found something hard to do, especially when others seem to do it with aplomb.

I have no doubt that there are swathes of good and decent people, but with the bile and defensive hatred that some cliques have, it's easy to see why some just don't bother engaging with the site at all.

Additionally, emotionless text doesn't help the situation, either - Most people reading this will have probably projected their internal expectations of frustration and turned my rather long comment into an 'angry rant' in their heads.

Much like YouTube needs quality content, I would say that so does GitHub. And, IMHO, GitHub is responsible for it's projects/content and it's husbandry.  Although GitHub doesn't make money from persuading and keeping people active on the site for as long as possible.  But, as has been said above, it is more of a repository than an entertainment site.

The OP has probably experienced, like me, a link on a website, purporting to be the best and most recent version of _thingumybob_1.1-i386_B_ you can get... I click the link, full of hope, but then I find myself at a GitHub page...  Where stores of useful information are kept in files have the _intuitive_ extension of _.md_...  Is this a file listing or a web page? Is this a text file? What is this strange place I find myself? Did I take a wrong turning somewhere?  I don't think I belong here.  I'd better leave.  But wait - I have no choice. If I want this piece of code which XXXX depends upon, I MUST navigate this strange hell known as 'GitHub'. I have no other choice.  GitHub or the highway.  How does this thing even work?! Then BOOM, a post like this appears on GitHub...  All I wanted was this file...

Maybe it is randommers linking to GitHub pages which shed no light on their purpose, but if GitHub can't quickly and easily communicate what it is to outsiders, it will always be a strange, niche, shadowy place to those not in the know

Just my shillings worth.

I totally understand where OP is coming from as I am not in this clique, either (I apparently don't ask questions correctly, or something pedantic.) I only come to GitHub to log bugs (there are so many bugs. I have wasted so much of my own time, it's quite depressing.  Some projects on GitHub have taught me to just leave and write my own code.  It actually got me back into programming after a 20+ year hiatus.) My fingers are tired and my brain is bent and broken... So many syntax errors and undeclared variables; But so little spare time...

**Comment by darismathew92** [comment]
> I get sent here a lot trying to get files but it's usually a dead end because I can't download anything.
> 
> WHY THE HELL DON'T YOU HAVE A SIMPLE "Download" button so that it's a million times easier to download files? I would never upload anything here, it's just way too confusing. And most of this stuff has no instructions either. Like I'm supposed to magically know how to install and use all these loose files I've never even seen before.
> 
> Please make this site more user friendly, like for people who don't know all this programming stuff.

Even programmers struggle at this. I have learned blockchain,JavaScript,..etc I enjoyed learning that but when it comes to github I totally hate it

**Comment by rodukov** [comment]
`git clone https://github.com/USERNAME/REPOSITORYNAME`
cd `REPOSITORYNAME`
This is all

**Reply by rodukov** [reply]
I forgot. You need to install git(if you did not do it yet)

**Comment by myConsciousness** [comment]
Just clone and don't download. It doesn't make sense for "Git"Hub.
